src/utils/narrow.js

```diff
@@ -390,6 +390,26 @@ export const emailsOfGroupPmNarrow = (narrow: Narrow): string[] =>
 export const emailsOfPmNarrow = (narrow: Narrow): string[] =>
   caseNarrowPartial(narrow, { pm: emails => emails });
 
+/**
+ * The stream name for a stream or topic narrow; else error.
+ *
+ * Most callers of this should probably be getting passed a stream name
+ * instead of a Narrow in the first place; or if they do handle other kinds
+ * of narrows, should be using `caseNarrow`.
+ */
+export const streamNameOfNarrow = (narrow: Narrow): string =>
```

Comment:
Nit: What's the pattern for how this and its neighbors are ordered? 🙂 I see three `is*Narrow`s (starting from `isHomeNarrow`), then these `*Of*Narrow`s...then seven more `is*Narrow`s.

I could see all the `is*Narrow`s being together in one section, and all the `*Of*Narrow`s being together in another section.

Or I could see the related ones being paired/grouped together. For example, `emailsOfGroupPmNarrow` and `emailsOfPmNarrow` are about PMs, so they could sit alongside `isGroupPmNarrow` and `is1to1PmNarrow`.

Reply:
Yeah. It's gotten a bit chaotic, in part because I've sometimes alternated between those two ways of organizing it as I've added things.

Also in part because I've considered several different ideas over time for what the eventual interface of this module and the `Narrow` type should be, after all this refactoring, and the bug-fixing it brings out, is complete. Some of these getters (like `emailsOfGroupPmNarrow`) get deleted already in later parts of my draft narrow-refactoring branch. My current thinking is that once `Narrow` has a reasonable structure, these getters (`*Of*Narrow`) and predicates (`is*Narrow`) will all simplify into trivial property accesses, and we can make the type a normal non-opaque type alias again (or perhaps better: an opaque type with a bound equal to its real underlying type, like `PmKeyRecipients`, so that outside code still can't *create* values of the type) and just inline all those getters and predicates at their call sites, and delete them.

So I'm inclined to defer any major reshuffling of the order of them until late in this series of PRs, when we have a firmer idea of what the set of things in the interface ultimately is.

Reply:
> My current thinking is that once `Narrow` has a reasonable structure, these getters (`*Of*Narrow`) and predicates (`is*Narrow`) will all simplify into trivial property accesses

That's great! 🤯 I wondered if something like that was on the horizon.